src/main/scala/cromwell/engine/db/slick/WorkflowExecutionComponent.scala

```diff
@@ -37,14 +37,30 @@ trait WorkflowExecutionComponent {
       workflowExecutionUuid, unique = true)
   }
 
-  val workflowExecutions = TableQuery[WorkflowExecutions]
+  protected val workflowExecutions = TableQuery[WorkflowExecutions]
 
   val workflowExecutionsAutoInc = workflowExecutions returning workflowExecutions.
     map(_.workflowExecutionId) into ((a, id) => a.copy(workflowExecutionId = Some(id)))
 
-  val workflowExecutionByID = Compiled(
-    (id: Rep[Int]) => for {
+  val workflowExecutionsByWorkflowExecutionUuid = Compiled(
+    (workflowExecutionUuid: Rep[String]) => for {
       workflowExecution <- workflowExecutions
-      if workflowExecution.workflowExecutionId === id
+      if workflowExecution.workflowExecutionUuid === workflowExecutionUuid
     } yield workflowExecution)
+
+  val workflowExecutionStatusesByWorkflowExecutionUuid = Compiled(
+    (workflowExecutionUuid: Rep[String]) => for {
+      workflowExecution <- workflowExecutions
+      if workflowExecution.workflowExecutionUuid === workflowExecutionUuid
+    } yield workflowExecution.status)
+
+  // NOTE: No precompile for you!
+  // [Compile] works for all functions ... consisting only of individual columns
```

Comment:
Added another link to [this thread](https://groups.google.com/forum/#!topic/scalaquery/2d_r4DEthfY) on the subject. Happy to be wrong though if someone knows something I don't. I'm still often feel like I'm ramping up a lot on this stuff too.